Userland/Libraries/LibArchive/TarStream.h

```diff
@@ -80,7 +80,8 @@ inline ErrorOr<void> TarInputStream::for_each_extended_header(F func)
 
     auto header_size = TRY(header().size());
     ByteBuffer file_contents_buffer = TRY(ByteBuffer::create_zeroed(header_size));
-    VERIFY(TRY(file_stream.read(file_contents_buffer)).size() == header_size);
+    if (TRY(file_stream.read(file_contents_buffer)).size() != header_size)
+        return Error::from_string_literal("Invalid size in header");
```

Comment:
This now looks `read_entire_buffer`-able:

```suggestion
    TRY(file_stream.read_entire_buffer(file_contents_buffer));
```